mempool/mempool.go

```diff
@@ -191,3 +191,29 @@ func (e ErrPreCheck) Error() string {
 func IsPreCheckError(err error) bool {
 	return errors.As(err, &ErrPreCheck{})
 }
+
+var ErrTxNotFound = errors.New("transaction not found in mempool")
+
+type ErrCheckTxAsync struct {
+	Reason error
```

Comment:
Personally, I would call this field just `Err`

Reply:
also, I think exported errors that wrap another `error` structure should have an `Unwrap() error` method that returns in underlying error. It is useful in many contexts, to give an example - it is useful when we want to check if an error has internally a specific error wrapped when writing unit tests

Reply:
Agreed, I'd recommend renaming `Reason` to `Err`.